apps/finance/app/src/components/Transfers.js

```diff
@@ -29,236 +30,220 @@ const TRANSFER_TYPES = [
-                )} to ${formatDate(selectedDateRange.end)}) selection. `}
-              {filtersActive && (
-                <a role="button" onClick={this.handleResetFilters}>
-                  Clear filters
-                </a>
-              )}
-            </p>
-          </NoTransfers>
-        ) : (
-          <div>
-            <Table
-              compactMode={compactMode}
-              header={
-                !compactMode && (
-                  <TableRow>
-                    <TableHeader title="Date" css="width: 12%" />
-                    <TableHeader title="Source / Recipient" css="width: 40%" />
-                    <TableHeader title="Reference" css="width: 100%" />
-                    <TableHeader title="Amount" align="right" css="width: 0" />
-                    <TableHeader />
-                  </TableRow>
-                )
-              }
-              css={`
-                color: ${theme.textPrimary};
-                margin-bottom: 20px;
-              `}
-            >
-              {filteredTransfers
-                .sort(({ date: dateLeft }, { date: dateRight }) =>
-                  // Sort by date descending
-                  compareDesc(dateLeft, dateRight)
-                )
-                .slice(0, displayedTransfers)
-                .map(transfer => (
-                  <TransferRow
-                    key={transfer.transactionHash}
-                    token={tokenDetails[toChecksumAddress(transfer.token)]}
-                    transaction={transfer}
-                    smallViewMode={compactMode}
-                  />
-                ))}
-            </Table>
-            {displayedTransfers < filteredTransfers.length && (
-              <Footer compactMode={compactMode}>
-                <Button mode="secondary" onClick={this.showMoreTransfers}>
-                  Show Older Transfers
-                </Button>
-              </Footer>
+  return (
+    <section>
+      <Header>
+        <Title compactMode={compactMode}>
+          <span>Transfers </span>
+          <span>
+            {compactMode && (
+              <ToggleFiltersButton
+                title="Toggle Filters"
+                onClick={handleToggleFiltersClick}
+                css="margin-right: -5px"
+              />
             )}
-          </div>
-        )}
-      </section>
-    )
-  }
-}
+          </span>
+        </Title>
+        <TransfersFilters
+          dateRangeFilter={selectedDateRange}
+          onDateRangeChange={setSelectedDateRange}
+          tokenFilter={selectedToken}
+          onTokenChange={handleTokenChange}
+          transferTypeFilter={selectedTransferType}
+          onTransferTypeChange={handleTransferTypeChange}
+          compactMode={compactMode}
+          opened={filtersOpened}
+          symbols={symbols}
+          transferTypes={TRANSFER_TYPES_STRING}
+          downloadFileName={getCsvFilename()}
+          onDownloadData={getDownloadData}
```

Comment:
Perhaps we could just hoist all of the download logic (name + data) into this component on an `handleDownloadTransactions()` callback?